src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ProjectFaultServiceExtensions.cs

```diff
@@ -0,0 +1,143 @@
+
+#pragma warning disable RS0030 // Do not used banned APIs (we are wrapping them)
+
+using System;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    /// <summary>
+    ///     Provides extensions methods for APIs in the project system that end up filing Watson reports.
+    /// </summary>
+    internal static class FaultExtensions
+    {
+        private static readonly ErrorReportSettings s_defaultReportSettings = new ErrorReportSettings(
+            eventName: "VisualStudioNonFatalErrors2",
+            component: "ManagedProjectSystem",
+            reportType: ErrorReportType.Critical,
+            submitFlags: ErrorReportSubmitFlags.OutOfProcess | ErrorReportSubmitFlags.NoCloseUI,
+            submitUIOptions: ImmutableDictionary.Create<ErrorReportUIType, string>());
+
+        /// <summary>
+        ///     Reports the specified fault.
+        /// </summary>
+        /// <param name="faultHandlerService">
+        ///     The <see cref="IProjectFaultHostHandler"/> that should handle the fault.
+        /// </param>
+        /// <param name="ex">
+        ///     Exception containing the fault information.
+        ///  </param>
+        /// <param name="severity">
+        ///     The severity of the failure.
+        /// </param>
+        /// <param name="project">
+        ///     The project related to the failure, if applicable. Can be <see langword="null"/>.
+        /// </param>
+        public static Task ReportFaultAsync(this IProjectFaultHandlerService faultHandlerService, Exception ex, UnconfiguredProject project, ProjectFaultSeverity severity = ProjectFaultSeverity.Recoverable)
+        {
+            Requires.NotNull(faultHandlerService, nameof(faultHandlerService));
+
+            return faultHandlerService.HandleFaultAsync(ex, s_defaultReportSettings, severity, project);
+        }
+
+        /// <summary>
+        ///     Attaches error handling to a task so that if it throws an unhandled exception,
+        ///     the error will be reported to the user.
+        /// </summary>
+        /// <param name="faultHandlerService">
+        ///     The <see cref="IProjectFaultHostHandler"/> that should handle the fault.
+        /// </param>
+        /// <param name="task">
+        ///     The task to attach error handling to.
+        /// </param>
+        /// <param name="project">
+        ///     The project related to the failure, if applicable. Can be <see langword="null"/>.
+        /// </param>
+        /// <param name="severity">
+        ///     The severity of the failure.
+        /// </param>
+        public static void Forget(this IProjectFaultHandlerService faultHandlerService, Task task, UnconfiguredProject project, ProjectFaultSeverity severity = ProjectFaultSeverity.Recoverable)
```

Comment:
Consider having `Forget` take `Func<Task>` so that exceptions thrown before yielding receive the same treatment as those thrown in continuations.

Reply:
I'm wrapping FaultHandlerService here, I don't control over this without reimplementing.